### ISSUE TITLE
Feature to upload a snapshot as soon as movement is detected

### ISSUE THREAD

**Issue by rodrigouroz** [ccrisan/motioneye#1332]
I want to fork the repo and create a new feature for uploading a snapshot (the first frame perhaps) when movement is detected. 
The idea is to upload a picture as quickly as possible when movement is detected, while the clip is being recorded and uploaded.
I tested MotionEye and loved it, however when movement is detected you have to wait for the clip to be finished and uploaded, I'm thinking on a situation where the hardware running this is tampered or disabled before the clip is uploaded. Uploading a picture should be faster.
I'm willing to implementing this myself and wanted to ask recommendations to people more used to this repo. Where should I start looking?

**Comment by ccrisan** [comment]
I don't think you need to fork anything: you can use the "Run A Command" option to execute a command whenever motion is detected. That corresponds to `on_event_start` Motion option.

**Comment by rodrigouroz** [comment]
Ok, it's not clear to me which info I'll get in the command from that event by looking at the code, I'll give it a try. I guess I could grab a frame from the detected camera as soon as the command is called.

**Comment by ccrisan** [comment]
@rodrigouroz you might need to look at [Motion](https://github.com/Motion-Project/motion) as well. Grabbing the first frame might be more difficult than it sounds.
